pic32/cores/pic32/WInterrupts.c

```diff
@@ -263,12 +265,22 @@ void __attribute__((nomips16,at_vector(_EXTERNAL_0_VECTOR),interrupt(IPL4SRS)))
 void __attribute__((interrupt(),nomips16)) ExtInt0Handler(void)
 #endif
 {
-
+	VEC_ENT_CODE(EXT_INT0,INT0IF,PIN_INT0);
```

Comment:
I have generated a couple of timing diagrams to better demonstrate my concern.  This is what I understand is how it is intended to work:

![isr](https://cloud.githubusercontent.com/assets/3812892/13605091/c8ae173a-e53d-11e5-98e7-5eb78906f940.png)

Whilst INT is low (for example) the interrupt will keep triggering.

This is my "concern" situation - where INT returns to HIGH *between* successive calls to the ISR. You get one extra call to the ISR with INT set to HIGH instead of set to LOW. At the time the ISR is called the INT pin is in a "not triggered" state. My concern is, should that interrupt actually execute the user's ISR? Since the test for whether IFS should be set or cleared isn't performed until the USER code is executed, and that USER code is executed regardless of the state of INT, is that actually the desired operation? 

![isr1](https://cloud.githubusercontent.com/assets/3812892/13605082/b7f1a786-e53d-11e5-8de8-dd114c0aeb5b.png)

All I am saying is, shouldn't there be a second check around the user code?

* Clear IFS
* Is the INT pin LOW?
  * Yes: execute user ISR
* Is the INT pin LOW?
  * Yes: Set IFS

That would result in this slight difference to the sequence of events:

![isr2](https://cloud.githubusercontent.com/assets/3812892/13605287/a6a3ca4e-e53e-11e5-9fad-0ad389c11bf1.png)

I don't doubt that the overall method is sound and reliable, my only concern is this one case of timing and should that fourth user ISR call in the second diagram actually exist, or should it be tested for and only executed if the INT pin is actually low at the time?